test/libsolidity/semanticTests/immutable/immutable_tag_too_large_bug.sol

```diff
@@ -0,0 +1,39 @@
+contract C {
```

Comment:
I tried to create a more concise test case, but I guess we need to have a somewhat convoluted code which can produce asm tags greater than 255...

Reply:
I am thinking that maybe this would be better as a cmdline test because the bug occurs at code generation at specific circunstances with `via-ir` and `optimize` enabled.

Reply:
If via-ir and optimize trigger it, then ``isoltest --optimize`` should also hit it on this case (if you add a bogus test call to it) - then it's fine.
And it'll be extremely hard to fabricate a minimal test case here - it's quite hard to achieve both a code size *very* close to 255, but also more than one reference to the same immutable that's not optimized away. So it's perfectly fine to use the OP's case.